pkg/controller/infrastructure/infraflow/ensure.go

```diff
@@ -91,16 +96,87 @@ func (fctx *FlowContext) ensureUserManagedVPC(ctx context.Context) error {
 	return nil
 }
 
-func (fctx *FlowContext) ensureSubnet(ctx context.Context) error {
-	var (
-		region = fctx.infra.Spec.Region
-	)
+func (fctx *FlowContext) ensureIPv6CIDRs(ctx context.Context) error {
+	nodeSubnet := fctx.whiteboard.GetObject(ObjectKeyNodeSubnet).(*compute.Subnetwork)
```

Comment:
Consider using: `GetObject[*compute.Subnetwork](fctx.whiteboard, ObjectKeyNodeSubnet)`
```suggestion
	nodeSubnet := GetObject[*compute.Subnetwork](fctx.whiteboard, ObjectKeyNodeSubnet)
```

Reply:
GetObject() checks if object exists and return new(T) in case if object is not found, the result will never be nil. But I need nil to check for the error case.